File: Node/utils/translate.text.js
const AWS = require('aws-sdk');
require('dotenv').config();

AWS.config.update({
    region: process.env.TRASLATE_REGION,
    accessKeyId:  process.env.TRASLATE_ACCESS_KEY_ID,
    secretAccessKey: process.env.TRASLATE_SECRET_ACCESS_KEY
});

const translate = new AWS.Translate({apiVersion: '2017-07-01'});

async function translateText(text, targetLanguage) {
    try {
        const params = {
            SourceLanguageCode: 'auto',
            TargetLanguageCode: targetLanguage,
            Text: text
        };
        const result = await translate.translateText(params).promise();
        return result.TranslatedText;
    } catch (err) {
        console.log(err);
        throw err;
    }
}

module.exports = translateText;

File: Python/utils/translateText.py
import boto3
import os
from dotenv import load_dotenv

load_dotenv()

translate = boto3.client('translate',
    aws_access_key_id=os.getenv('TRANSLATE_ACCESS_KEY_ID'),
    aws_secret_access_key=os.getenv('TRANSLATE_SECRET_ACCESS_KEY'),
    region_name=os.getenv('TRANSLATE_REGION')
)

def translate_text(text, target_language):
    try:
        params = {
            'SourceLanguageCode': 'auto',
            'TargetLanguageCode': target_language,
            'Text': text
        }

        result = translate.translate_text(**params)
        return result.get('TranslatedText')
    
    except Exception as e:
        print(f"Error al traducir el texto: {e}")
        raise e